### PR TITLE
Upgrade to AWS SDK V2

### PR DESCRIPTION
**Pull request by CallumNZ**
**Reference:**

https://github.com/GeoNet/tickets/issues/10456, https://github.com/GeoNet/tickets/issues/11752, https://github.com/GeoNet/tickets/issues/11780

**Changes:**

- Upgrade to Go AWS SDK V2
- Remove usage of old pkg/errors
- Upgrade go version to 1.16
- Add timeout for server connections

**Testing (ones with AWS changes)**

- [x] fdsn-holdings-consumer
- [x] fdsn-quake-consumer
- [x] fdsn-ws
- [ ] s3-notify

### CODE REVIEW

**Comment by CallumNZ** [comment]
@junghao Deployed to dev. I ran .`/build.sh` for the 4 applications mentioned above successfully. Then ran `./push.sh` for the 4, which worked for all except `s3-notify`.

Do I need to do something special for `s3-notify`?
To see if everything's working, do we wait for a new earthquake to come through?

**Comment by junghao** [comment]
s3 notify is a command line tool are rarely used, so I don't think we need to test it live.

File: cmd/s3-notify/s3-notify.go [hunk 1]
```diff
@@ -4,12 +4,12 @@ import (
...
-	fdsnS3 "github.com/GeoNet/fdsn/internal/platform/s3"
-	"github.com/GeoNet/fdsn/internal/platform/sqs"
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/s3"
 	"log"
+	"strings"
+
+	"github.com/GeoNet/kit/aws/s3"
+	fdsnS3 "github.com/GeoNet/kit/aws/s3"
```
Comment (junghao) [inline cmd/s3-notify/s3-notify.go:11]: Redundant?

**Review by junghao — COMMENTED** [review]
Shall we make it go 1.16?

**Comment by CallumNZ** [comment]
I have made those changes and redeployed `fdsn-holdings-consumer` , `fdsn-quake-consumer`, and `fdsn-ws` to dev.

**Review by junghao — APPROVED** [review]
Tried to deploy in dev environment for fdsn-ws, fdsn-quake-consumer, fdsn-holdings-consumer, seems running well (no unexpected error logs, consumes SQS messages correctly).

**Pull request merged**